frontend/packages/console-app/src/components/modals/restore-pvc/restore-pvc-modal.tsx

```diff
@@ -165,16 +170,23 @@ const RestorePVCModal = withHandlePromise<RestorePVCModalProps>(
               label="Size"
               isRequired
               fieldId="pvc-size"
-              className="co-restore-pvc-modal__input"
+              className="co-restore-pvc-modal__input co-restore-pvc-modal__ocs-size"
+              helperTextInvalid="Size should be equal or higher than the restore size of snapshot"
+              validated={validSize ? 'default' : 'error'}
             >
-              <RequestSizeInput
-                name="requestSize"
-                onChange={requestedSizeInputChange}
-                defaultRequestSizeUnit={requestedUnit}
-                defaultRequestSizeValue={requestedSize}
-                dropdownUnits={dropdownUnits}
-                required
-              />
+              {snapshotClassResourceLoaded && !snapshotClassResourceLoadError ? (
+                <RequestSizeInput
+                  name="requestSize"
+                  onChange={requestedSizeInputChange}
+                  defaultRequestSizeUnit={requestedUnit}
+                  defaultRequestSizeValue={requestedSize}
+                  dropdownUnits={dropdownUnits}
+                  isInputDisabled={isCephProvisioner(snapshotClassResource?.driver)}
+                  required
+                />
+              ) : (
+                <div className="skeleton-text" />
```

Comment:
same as above

Reply:
Good point, its better to disable size input in that case.